### PR TITLE
🐛 Source S3: Make `Advanced Reader Options` and `Advanced Options` truly `Optional`

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving: https://github.com/airbytehq/oncall/issues/1585

## 🚨 User Impact 🚨
No impact is expected.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

**Review by ryjabe — APPROVED** [review]
I don't think I am qualified to review, so probably want someone else to do it.  That said from my limited perspective this looks good.  Thank you for the quick action!

**Comment by bazarnov** [comment]
> I don't think I am qualified to review, so probably want someone else to do it. That said from my limited perspective this looks good. Thank you for the quick action!

No worries, you've been assigned here just because you're the issue reporter)

File: airbyte-integrations/connectors/source-s3/source_s3/utils.py [hunk 1]
```diff
@@ -50,5 +50,5 @@ def multiprocess_queuer(func: Callable, queue: mp.Queue, *args: Any, **kwargs: A
     queue.put(dill.loads(func)(*args, **kwargs))
 
 
-def get_value_or_json_if_empty_string(options: str) -> str:
-    return options.strip() or "{}"
+def get_value_or_json_if_empty_string(options: str = None) -> str:
+    return options.strip() if options else "{}"
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-s3/source_s3/utils.py:54]: 👍 does what it says!  Replace null options with empty object. 
Screenshot 2023-03-02 at 2 15 23 PM

**Review by evantahler — APPROVED** [review]

File: airbyte-integrations/connectors/source-s3/source_s3/utils.py (hunk 1 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-s3/source_s3/utils.py:54]: The options could contain None or "None", which will cause issues is the future. Leaving it as it is.

**Pull request merged**